x/distribution/keeper/delegation.go

```diff
@@ -87,18 +92,26 @@ func (k Keeper) calculateDelegationRewards(ctx sdk.Context, val sdk.Validator, d
 		)
 	}
 
-	// a stake sanity check - recalculated final stake should be less than or equal to current stake
-	// here we cannot use Equals because stake is truncated when multiplied by slash fractions
-	// we could only use equals if we had arbitrary-precision rationals
+	// A total stake sanity check; Recalculated final stake should be less than or
+	// equal to current stake here. We cannot use Equals because stake is truncated
+	// when multiplied by slash fractions (see above). We could only use equals if
+	// we had arbitrary-precision rationals.
 	currentStake := val.TokensFromShares(del.GetShares())
 	if stake.GT(currentStake) {
-		panic(fmt.Sprintf("calculated final stake for delegator %s greater than current stake: %s, %s",
-			del.GetDelegatorAddr(), stake, currentStake))
+		// account for rounding errors due to stake being multiplied by slash fractions
```

Comment:
Why would these rounding errors lead to `stake` being too high, if we round *up* the slash fractions?